### ISSUE TITLE
Package dependencies not compatible

### ISSUE THREAD

**Issue by janbruedigam** [JuliaReinforcementLearning/ReinforcementLearning.jl#860]
I might be missing something, but when adding ReinforcementLearning, ReinforcementLearningBase, ReinforcementLearningCore in an empty environment, I get the following versions:
   [158674fc] + ReinforcementLearning v0.10.2
⌅ [e575027e] + ReinforcementLearningBase v0.9.7
⌅ [de1b191a] + ReinforcementLearningCore v0.8.13

I would imagine that the packages in the ReinforcementLearning.jl repo work with each other. But Base and Core can't be updated to newer versions, which breaks some examples for me.
Probably also related to https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/issues/781  

Using Julia 1.8.5

**Comment by jeremiahpslewis** [comment]
https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/issues/792

The package is in the midst of a refactor which means there are two 'most recent' versions: one if you just use the code packages (see the ones which are tested in CI/CD here (https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/e2da341a12d74edaa5348866684c02421f77c1d4/.github/workflows/ci.yml#L99) and one which is the set of versions you get when you install the ReinforcementLearning.jl Meta-Package. Once the sub-packages are once again feature complete (e.g. can run all or at least most of the examples) then a new version of ReinforcementLearning.jl can be tagged with updated compat bounds.

If you're actively using the package and would like to contribute to the refactor, let me know and I can try and point you in the right direction. :)

**Comment by janbruedigam** [comment]
Ah ok, thanks for the explanation! 
We're just updating the examples in https://github.com/dojo-sim/Dojo.jl.
I have/had an example working which uses Dojo's cartpole dynamics for your `CartPoleEnv` and `BasicDQNLearner` example. But then we'll wait a bit with interfacing RL.jl properly until your updates are done.
Just in case I missed it, this example hasn't been updated yet, right? The current one uses apparently outdated things like `NeuralNetworkApproximator` or `CircularArraySARTTrajectory` and I'm not sure how they got replaced exactly.

**Comment by jeremiahpslewis** [comment]
@HenriDeh any thoughts about this?

**Comment by HenriDeh** [comment]
I think `NeuralNetworkApproximator` is now `Approximator` and for the replacement of `CircularArraySARTTrajectory` you will have to use a `Trajectory` from `ReinforcemenLearningtTrajectories.jl` (exported by RL.jl). You can checkout [this working experiment](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/main/src/ReinforcementLearningExperiments/deps/experiments/experiments/Policy%20Gradient/JuliaRL_MPO_CartPole.jl) to see how to use all of that. 

I don't know why your Base and Core packages are the latest, but these are pretty old versions. Are you sure it is not another dependency from your environment that constraints the versions? You can see that by doing `]up --outdated -m`

**Comment by janbruedigam** [comment]
Thanks, I'll have a look at the example.

On the second part (`up` doesn't have an option `outdated`, but `status` works):
```
(jl_C1Nl5i) pkg> status
Status `C:\Users\Jan\AppData\Local\Temp\jl_C1Nl5i\Project.toml`
  [158674fc] ReinforcementLearning v0.10.2
⌅ [e575027e] ReinforcementLearningBase v0.9.7
⌅ [de1b191a] ReinforcementLearningCore v0.8.13
Info Packages marked with ⌅ have new versions available but compatibility constraints restrict them from upgrading. To see why use `status --outdated`

(jl_C1Nl5i) pkg> status --outdated -m
Status `C:\Users\Jan\AppData\Local\Temp\jl_C1Nl5i\Manifest.toml`
⌅ [1520ce14] AbstractTrees v0.3.4 (<v0.4.4): Flux, ReinforcementLearningBase, ReinforcementLearningCore, ReinforcementLearningZoo
⌅ [4fba245c] ArrayInterface v5.0.8 (<v7.4.3): EllipsisNotation, Flux, ReinforcementLearningCore
⌅ [ab4f0b2a] BFloat16s v0.2.0 (<v0.4.2): CUDA
⌅ [052768ef] CUDA v3.13.1 (<v4.1.3): Flux, NNlibCUDA, ReinforcementLearningCore, ReinforcementLearningZoo
⌅ [34da2185] Compat v3.46.2 (<v4.6.1): ArrayInterface, ReinforcementLearningCore
⌅ [d38c429a] Contour v0.5.7 (<v0.6.2): UnicodePlots
⌃ [da5c29d0] EllipsisNotation v1.6.0 (<v1.7.0)
⌅ [1a297f60] FillArrays v0.13.11 (<v1.0.0): ReinforcementLearningCore
⌅ [587475ba] Flux v0.12.10 (<v0.13.14): ReinforcementLearningCore
⌅ [663a7486] FreeTypeAbstraction v0.9.9 (<v0.10.0): UnicodePlots
⌅ [d9f16b24] Functors v0.2.8 (<v0.4.4): Flux, ReinforcementLearningCore
⌅ [61eb1bfa] GPUCompiler v0.17.3 (<v0.19.0): CUDA
⌅ [8197267c] IntervalSets v0.5.4 (<v0.7.4): ReinforcementLearningEnvironments, ReinforcementLearningZoo
⌅ [929cbde3] LLVM v4.17.1 (<v5.0.0): CUDA, GPUCompiler
⌃ [a00861dc] NNlibCUDA v0.2.6 (<v0.2.7)
⌅ [e575027e] ReinforcementLearningBase v0.9.7 (<v0.11.0): ReinforcementLearning, ReinforcementLearningCore, ReinforcementLearningEnvironments, ReinforcementLearningZoo
⌅ [de1b191a] ReinforcementLearningCore v0.8.13 (<v0.9.2): ReinforcementLearning, ReinforcementLearningZoo
⌅ [25e41dd2] ReinforcementLearningEnvironments v0.6.12 (<v0.7.1): ReinforcementLearning
⌅ [d607f57d] ReinforcementLearningZoo v0.5.12 (<v0.6.0): ReinforcementLearning
⌅ [efcf1570] Setfield v0.8.2 (<v1.1.1): ReinforcementLearningCore, ReinforcementLearningZoo
⌅ [aedffcd0] Static v0.6.6 (<v0.8.6): ArrayInterface
⌅ [b8865327] UnicodePlots v2.12.4 (<v3.5.1): ReinforcementLearningCore
⌅ [a5390f91] ZipFile v0.9.4 (<v0.10.1): Flux
⌅ [dad2f222] LLVMExtra_jll v0.0.18+0 (<v0.0.21+0): LLVM
```

**Comment by jeremiahpslewis** [comment]
Remove ReinfocementLearning.jl and update your environment and you should have the latest dependencies. As RL.jl is just a meta package, the only change necessary is updating your import / using statements. Once the next RL.jl version is released, you can revert this.

**Comment by jeremiahpslewis** [comment]
👍 ok, just checking that things are (at least) working as expected. definitely lots of breaking changes, which will continue a bit until things are stabilized

**Comment by HenriDeh** [comment]
Oh right the Agent thing I had to fix in #849. It should work on master but for unknown reasons it passed it 0.10.2 tests but started failing again. We may need a new release.

**Comment by jeremiahpslewis** [comment]
I believe this is fixed.
